### PR TITLE
apache drill grammar

### PR DESCRIPTION
**Pull request by mlorek**

### CODE REVIEW

**Comment by mlorek** [comment]
@kaby76 how can I see more details about the following:
`DrillParser.ts(7245,9): error TS2393: Duplicate function implementation.`

**Comment by kaby76** [comment]
> @kaby76 how can I see more details about the following: `DrillParser.ts(7245,9): error TS2393: Duplicate function implementation.`

For the typescript test.sh, it looks like I didn't redirect correctly. I will check.

```
03/09-10:49:07 ~/fb/grammars-v4/sql/drill/Generated-TypeScript
$ make
bash build.sh
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

changed 19 packages, and audited 20 packages in 3s

found 0 vulnerabilities
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

added 11 packages, and audited 12 packages in 2s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

> i@1.0.0 build
> tsc -p tsconfig.json

DrillParser.ts:7245:9 - error TS2393: Duplicate function implementation.

7245  public expression_list(): ExpressionContext[] {
             ~~~~~~~~~~~~~~~

DrillParser.ts:7296:9 - error TS2393: Duplicate function implementation.

7296  public expression_list(): Expression_listContext {
             ~~~~~~~~~~~~~~~

DrillParser.ts:7378:9 - error TS2393: Duplicate function implementation.

7378  public expression_list(): ExpressionContext[] {
             ~~~~~~~~~~~~~~~

DrillParser.ts:7435:9 - error TS2393: Duplicate function implementation.

7435  public expression_list(): Expression_listContext {
             ~~~~~~~~~~~~~~~


Found 4 errors in the same file, starting at: DrillParser.ts:7245

mingw32-make: *** [makefile:3: build] Error 2
```

**Comment by kaby76** [comment]
Looks like the all wise, all knowing, great programmers of tsc decided to output the error messages from the compiler to something other than stderr and stdout. I'm not sure yet how I can redirect TypeScript error messages.

**Comment by kaby76** [comment]
Instead of disabling TypeScript, you can just rename "expression_list" to "expression_list_". It's a "symbol conflict".

File: sql/drill/DrillLexer.g4 [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+COMMA             : ',';
+SEMI              : ';';
+STAR              : '*';
+DIVIDE            : '/';
+MODULE            : '%';
+PLUS              : '+';
+MINUS             : '-';
+
+fragment LETTER      : [A-Z_];
+fragment DEC_DOT_DEC : (DEC_DIGIT+ '.' DEC_DIGIT+ |  DEC_DIGIT+ '.' | '.' DEC_DIGIT+);
```
Comment (KvanTTT) [inline sql/drill/DrillLexer.g4:190]: Useless parentheses:

```suggestion
fragment DEC_DOT_DEC : DEC_DIGIT+ '.' DEC_DIGIT+ |  DEC_DIGIT+ '.' | '.' DEC_DIGIT+;
```

File: sql/drill/DrillLexer.g4 [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+IDENTIFIER        : [A-Z_] [A-Z0-9_]*;
+
+BS_STRING_LITERAL : '`'  ~'`'+ '`';
+SQ_STRING_LITERAL : '\'' ~'\''+ '\'';
+DQ_STRING_LITERAL : '"' ~'"'+ '"';
+
+DECIMAL_LITERAL   : DEC_DIGIT+;
+FLOAT_LITERAL     : DEC_DOT_DEC;
+REAL_LITERAL      : (DECIMAL_LITERAL | DEC_DOT_DEC) ('E' [+-]? DEC_DIGIT+);
+CHAR_LITERAL      : '\'' (~['\\\r\n]) '\'';
```
Comment (KvanTTT) [inline sql/drill/DrillLexer.g4:162]:
```suggestion
CHAR_LITERAL      : '\'' ~['\\\r\n] '\'';
```

File: sql/drill/DrillLexer.g4 [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+IDENTIFIER        : [A-Z_] [A-Z0-9_]*;
+
+BS_STRING_LITERAL : '`'  ~'`'+ '`';
+SQ_STRING_LITERAL : '\'' ~'\''+ '\'';
+DQ_STRING_LITERAL : '"' ~'"'+ '"';
+
+DECIMAL_LITERAL   : DEC_DIGIT+;
+FLOAT_LITERAL     : DEC_DOT_DEC;
+REAL_LITERAL      : (DECIMAL_LITERAL | DEC_DOT_DEC) ('E' [+-]? DEC_DIGIT+);
```
Comment (KvanTTT) [inline sql/drill/DrillLexer.g4:161]:
```suggestion
REAL_LITERAL      : (DECIMAL_LITERAL | DEC_DOT_DEC) 'E' [+-]? DEC_DIGIT+;
```

File: sql/drill/DrillParser.g4 [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+reset_command
+    : (ALTER SESSION)? RESET id_
+    ;
+
+refresh_table_metadata
+    : REFRESH TABLE METADATA (COLUMNS column_list_paren | NONE)? table_name
+    ;
+
+analyze_command
+    : ANALYZE TABLE (table_name | TABLE'(' id_'(' param_list ')' ')')?
```
Comment (KvanTTT) [inline sql/drill/DrillParser.g4:159]: Missing spaces:

```suggestion
    : ANALYZE TABLE (table_name | TABLE '(' id_ '(' param_list ')' ')')?
```

**Review by KvanTTT — CHANGES_REQUESTED** [review]
Please remove redundant parentheses.

**Comment by kaby76** [comment]
The problem with redirecting the output from the TypeScript is solved using "tsc --pretty -p tsconfig.json". The fix will be in with https://github.com/antlr/grammars-v4/pull/3195

**Comment by mlorek** [comment]
@KvanTTT all ok?

**Review by KvanTTT — APPROVED** [review]

**Pull request merged**